src/Common/src/Microsoft/Win32/SafeHandles/NtlmSecuritySafeHandles.cs

```diff
@@ -0,0 +1,271 @@
+        }
+
+        public byte[] GetResponse(uint flags, string username, string password, string domain,
+                                  out byte[] sessionKey)
+        {
+            // reference for NTLM response: https://msdn.microsoft.com/en-us/library/cc236700.aspx
+            sessionKey = null;
+            SafeNtlmBufferHandle key;
+            int keyLen;
+            int status = Interop.NetSecurity.HeimNtlmNtKey(password, out key, out keyLen);
+            Interop.NetSecurity.HeimdalNtlmException.AssertOrThrowIfError("HeimNtlmKey failed", status);
+
+            using (key)
+            {
+                byte[] baseSessionKey = new byte[Interop.NetSecurity.MD5DigestLength];
+                SafeNtlmBufferHandle lmResponse;
+                int lmResponseLength;
+
+                status = Interop.NetSecurity.HeimNtlmCalculateResponse(true, key, _type2Handle, username, domain,
+                         baseSessionKey, baseSessionKey.Length, out lmResponse, out lmResponseLength);
+                Interop.NetSecurity.HeimdalNtlmException.AssertOrThrowIfError("HeimNtlmCalculateResponse lm1 failed",status);
+
+                SafeNtlmBufferHandle ntResponse;
+                int ntResponseLength;
+                status = Interop.NetSecurity.HeimNtlmCalculateResponse(false, key, _type2Handle, username, domain,
+                                                                       baseSessionKey, baseSessionKey.Length, out ntResponse, out ntResponseLength);
+                Interop.NetSecurity.HeimdalNtlmException.AssertOrThrowIfError("HeimNtlmCalculateResponse lm2 failed", status);
+
+                SafeNtlmBufferHandle sessionKeyHandle = null;
+                int sessionKeyLen;
+                SafeNtlmBufferHandle outputData = null;
+                int outputDataLen = 0;
+                status = Interop.NetSecurity.CreateType3Message(key, _type2Handle, username, domain, flags, lmResponse, ntResponse, baseSessionKey,
+                                                                baseSessionKey.Length, out sessionKeyHandle, out sessionKeyLen, out outputData,
+                                                                out outputDataLen);
+                Interop.NetSecurity.HeimdalNtlmException.AssertOrThrowIfError("CreateType3Message failed", status);
+                using (sessionKeyHandle)
+                using (outputData)
+                {
+                    sessionKey = sessionKeyHandle.ToByteArray(sessionKeyLen,0);
+                    return outputData.ToByteArray(outputDataLen, 0);
+                }
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _type2Handle.Dispose();
+                _type2Handle = null;
+            }
+            base.Dispose(disposing);
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            return true;
```

Comment:
Same question @bartonjs had in another place... if there's nothing to actually release, why is this a SafeHandle?